[Daniel Lim Wee Soong] iP

daniellimws opened the pull request:
Add basic functionality for Duke to:

* Start chatting with the user by showing his face
* List user's tasks
* Allow user to add tasks of multiple types
* Mark user's task as done
* Graciously shutdown when the user says bye

daniellimws committed:
Add welcome and exit message

daniellimws committed:
Improve greeting and respond to user input

- Currently only echos the user input, or exits the program if user says
"bye"
- Change duke logo to robot head

daniellimws committed:
Implement storing of user input in an array

daniellimws committed:
Add Task class to track task status

daniellimws committed:
Support different types of tasks

Allow user to specify the type of tasks by starting each command with
one of (Todo/Deadline/Event)

daniellimws committed:
Refactor constants and ensure SLAP

daniellimws committed:
Add test cases

esskayzz reviewed (commented):
Follows coding standards. Constants are in caps and variables in Pascalcase and functions in camelcase.

alstontham commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,149 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String logo =
```
Perhaps it would be better to put your welcome message and the logo into a method of its own

daniellimws replied:
Done

alstontham reviewed (commented):
Overall your code looks great! It was very easy to follow and was intuitive. The only comment I could make is to put your welcome message and logo into a method of its own. Other than that great job!

esskayzz commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,149 @@
+        }
+    }
+
+    private static void markTaskDone(String cmd) {
```
Good use of naming and follows naming convention. It would be good to add a few comments for readability.

alstontham commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,149 @@
+        printResponse(String.format("Added: %s", task));
+        tasks[taskCount++] = task;
+        printResponse(
+                String.format(
```
I think you could format line 102, 103, and 104 to be on the same line since it doesn't exceed the line limit of 120 characters for a single line.

daniellimws replied:
Thanks. I've updated accordingly.

alstontham reviewed (commented):
Just found a minor formatting point that could be changed, but everything else looks good!

daniellimws committed:
Organize source files into duke package

daniellimws committed:
Refactor methods in Duke and improve code formatting

daniellimws committed:
Update runtest script to use duke.Duke

daniellimws committed:
Handle errors due to wrong user input

esskayzz commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,32 @@
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
```
Perhaps a good name for your method can be setAsDone() as stipulated in the coding standard

daniellimws replied:
Changed, thanks.

esskayzz commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,243 @@
+        tasks[taskCount++] = task;
+        printResponse(String.format("Added: %s", task));
+        printResponse(String.format("You now have %d task%s in your list.", taskCount, (taskCount == 1 ? "" : "s")));
+    }
```
Did you check if your taskCount exceeds your task capacity? And the use ternary operator makes the logic of the code hard to read.

daniellimws replied:
Fixed, thanks.

daniellimws committed:
Merge branch 'branch-Level-5' into master

daniellimws committed:
Check if task_count exceeds capacity and improve code readability

daniellimws committed:
Add support for deleting of tasks

Change implementation to use ArrayList<Task> instead of Task[]

daniellimws committed:
Auto save tasks data

0WN463 commented on data/duke.txt:
```diff
@@ -0,0 +1,2 @@
+T,N,borrow
+E,N,aaa,aaa
```
Ideally, this file should be ignored using the .gitignore file, so that git does not track such trivial changes

daniellimws replied:
Oh thanks

0WN463 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,336 @@
+import java.util.Scanner;
+
+public class Duke {
+    private static final String AT_ARGUMENT = "/at ";
+    private static final String BY_ARGUMENT = "/by ";
+
+    private static final String BYE_COMMAND = "bye";
+    private static final String DONE_COMMAND = "done";
+    private static final String LIST_COMMAND = "list";
+    private static final String DELETE_COMMAND = "delete";
+
+    private static final String EVENT_COMMAND = "event";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String TODO_COMMAND = "todo";
+
+    private static final String PROMPT = "> ";
+    private static final String PRE_RESPONSE_WHITESPACE = "  ";
+
+    private static ArrayList<Task> tasks = new ArrayList<Task>();
+
+    private static final String DATA_DIR = "data";
+    private static final String SAVE_PATH = "data/duke.txt";
+
```
Good use of constants

0WN463 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,336 @@
+        String description, date;
+
+        try {
+            description = args.substring(0, dateIndex - 1);
+            date = args.substring(dateIndex + BY_ARGUMENT.length());
```
Good use of temporary variables as labels

0WN463 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,336 @@
+        loadData();
+        welcome();
+
+        Scanner in = new Scanner(System.in);
```
Consider using a more descriptive variable name

0WN463 commented on src/main/java/duke/exception/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,6 @@
+
+public class EmptyDescriptionException extends DukeException {
+
+    private static final long serialVersionUID = 4037951695361980047L;
```
I'm not sure what is the serialVersionUID? Did you use some autogeneration tool to create this class?

daniellimws replied:
Yea I was using vscode and it recommended this. I removed it since it wasn't necessarily.

0WN463 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,25 @@
+        return String.format("[D]%s (by: %s)", super.toString(), date);
+    }
+
+    @Override
+    protected String getSerializeType() {
+        return "D";
+    }
+
```
Good use of polymorphism to help compartmentalize code

0WN463 reviewed (commented):
Overall, very well done.

Only minor nitpicks here and there

daniellimws committed:
Add data/ to gitignore

daniellimws committed:
Using better variable name for stdin scanner

daniellimws committed:
Refactor code by extracting out closely related code

`Ui`: Deals with interactions with the user
`Storage`: Deals with loading tasks from the file and saving tasks in
the file
`Parser`: Deals with making sense of the user command
`TaskList`: Contains the task list

daniellimws committed:
Use LocalDate class to store dates in tasks

daniellimws committed:
Implement find command to filter tasks

daniellimws committed:
Merge pull request #2 from daniellimws/branch-Level-8

Level-8: Use LocalDate class to store dates in tasks

daniellimws committed:
Merge pull request #1 from daniellimws/branch-Level-9

Level 9: Implement find command to filter tasks

daniellimws committed:
Add JavaDoc comments

daniellimws committed:
Merge pull request #3 from daniellimws/branch-A-JavaDoc

A-JavaDoc: Branch a java doc

daniellimws committed:
Update help messages for event and deadline

daniellimws committed:
Add newline add the end of help messages

daniellimws committed:
Parser: Show help message when no argument is given to find

daniellimws committed:
Add user guide

daniellimws committed:
Code cleanup

daniellimws committed:
Escape comma in task description

daniellimws committed:
Remove redundant method
